### DISCUSSION TITLE
Filter API rule from map

### DISCUSSION THREAD

**Discussion by Oggy4th** [pocketbase/pocketbase#2655] [Q&A] [answered]
Is it possible to filter from map?

**Comment by ganigeorgiev** [comment]
I'm not sure I follow. Could you elaborate and/or provide an example?

**Reply by Oggy4th** [reply]
There is a json map in my record that I would like to filter on in api rules
```
{
  "h24zr9qgglro0lk": {
    "UPDATE_RESTOCK": false
  },
  "4kxqve2w3nturt3": {
    "CREATE_RESTOCK": true
  }
}
```
I'm trying to only fetch the records that has map corresponding with id of `4kxqve2w3nturt3` and if `CREATE_RESTOCK` equals to `true`

**Reply by ganigeorgiev** [reply, marked as answer]
We have basic [`json_extract`](https://www.sqlite.org/json1.html#jex) support using dot-notation, meaning that you can try something like:

```
someJsonField.4kxqve2w3nturt3.CREATE_RESTOCK = true
```

**Reply by Oggy4th** [reply]
it works but is it possible to use @request.auth.id instead of the id?

**Reply by ganigeorgiev** [reply]
No, the filter path can't have dynamic elements at the moment.

Depending on your use case, one possible workaround could be to refactor the above json object to its own collection (eg. with separate `user` relation field).

**Reply by Oggy4th** [reply]
Ok I'll refactor it with relation fields, thank you very much
